tests/Bootstrap/ListGroupTest.elm

```diff
@@ -0,0 +1,210 @@
+module Bootstrap.ListGroupTest exposing (..)
+
+import Bootstrap.ListGroup as ListGroup
+import Html exposing (text)
+import Html.Attributes as Attr exposing (href)
+import Test exposing (Test, test, describe)
+import Expect
+import Test.Html.Query as Query
+import Test.Html.Selector as Selector exposing (tag, class, classes, attribute, disabled)
+
+
+{-| @ltignore
+-}
+all : Test
+all =
+    Test.concat
+        [ vanillaListGroup
+        , keyedListGroup
+        , contextualListGroup
+        , customListGroup
+        , contextualListGroup
+        ]
+
+
+vanillaListGroup : Test
+vanillaListGroup =
+    let
+        html =
+            ListGroup.ul
+                [ ListGroup.li [] [ text "List item 1" ]
+                ]
+
+        active =
+            ListGroup.ul
+                [ ListGroup.li [ ListGroup.active ] [ text "List item 1" ]
+                ]
+
+        disabled =
+            ListGroup.ul
+                [ ListGroup.li [ ListGroup.disabled ] [ text "List item 1" ]
+                ]
+    in
+        describe "vanilla ListGroup"
+            [ test "expect ul with list-group class" <|
+                \() ->
+                    html
+                        |> Query.fromHtml
+                        |> Query.has [ tag "ul", class "list-group" ]
+            , test "expect li with list-group-item class" <|
+                \() ->
+                    html
+                        |> Query.fromHtml
+                        |> Query.has [ tag "li", class "list-group-item" ]
+            , test "expect li is not disabled" <|
+                \() ->
+                    html
+                        |> Query.fromHtml
+                        |> Query.has [ tag "li", Selector.disabled False ]
+            , test "expect li has content" <|
+                \() ->
+                    html
+                        |> Query.fromHtml
+                        |> Query.has [ tag "li", Selector.text "List item 1" ]
+            , test "expect li with active class" <|
+                \() ->
+                    active
+                        |> Query.fromHtml
+                        |> Query.has [ tag "li", class "active" ]
+            , test "expect li with disabled class and attribute" <|
+                \() ->
+                    disabled
+                        |> Query.fromHtml
+                        |> Query.has [ tag "li", class "disabled", Selector.disabled True ]
+            ]
+
+
+{-| The fact that an element is keyed is not visible from the dom, so
+this just tests that the keyed versions produce the same html as the "vanilla" ones
+-}
+keyedListGroup : Test
```

Comment:
Yeah, I think it might be worth extracting all that's common between vanilla and keyed into a separate function used by both.